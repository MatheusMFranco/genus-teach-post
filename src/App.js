import logo from './ghost.jfif';
import './App.less';

function App() {
  return (
    <article className="quote">
      <div className="quote-media">
        <img src={logo} className="quote-picture" alt="Zeitgeist" />
      </div>
      <blockquote className="quote-box">
        <h1 className="quote-message">
          Quote
        </h1>
        <cite className="quote-author">
        Author
        </cite>
      </blockquote>
    </article>
  );
}

export default App;
