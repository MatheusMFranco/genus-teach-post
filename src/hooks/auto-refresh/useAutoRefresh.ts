import { useEffect } from 'react';

export const useAutoRefresh = (interval: number) => {
    const TIME = 60000;
    const minutes = interval * TIME;
    useEffect(() => {
        const intervalId = setInterval(() => window.location.reload(), minutes);
        return () => clearInterval(intervalId);
    }, [minutes]);
};
