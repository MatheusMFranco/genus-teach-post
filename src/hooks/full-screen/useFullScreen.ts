import { useState, useCallback } from 'react';

export const useFullScreen = () => {
    const [isFullScreen, setIsFullScreen] = useState(!!document.fullscreenElement);

    const toggleFullScreen = useCallback(() => {
        /* istanbul ignore else */
        if (!isFullScreen) {
            document.documentElement.requestFullscreen();
        } else {
            document.exitFullscreen();
        }
        setIsFullScreen((prev) => !prev);
    }, [isFullScreen]);

    return { toggleFullScreen, isFullScreen };
};
