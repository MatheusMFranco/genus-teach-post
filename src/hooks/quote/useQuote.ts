import { useState, useEffect } from 'react';

import { Quote } from '../../models/Quote';
import { fetchRandomQuote } from '../../services/random/randomService';
import { fetchAlternativeQuote } from '../../services/alternative/alternativeService';

export const useQuote = () => {
  const [message, setMessage] = useState({} as Quote);

  useEffect(() => {
    const fetchQuote = async () => {
      try {
        const quote = await fetchRandomQuote();
        setMessage(quote);
      } catch (error) {
        try {
          const alternativeQuote = await fetchAlternativeQuote();
          setMessage(alternativeQuote);
        } catch (error) {
          setMessage({
            quote: "I'm sorry, Dave. I'm afraid I can't do that.",
            author: 'Hal 9000',
          });
        }
      }
    };
    fetchQuote();
  }, []);

  return message;
};
