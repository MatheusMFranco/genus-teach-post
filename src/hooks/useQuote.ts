import { useState, useEffect } from 'react';

import { Quote } from '../models/Quote';
import { fetchRandomQuote } from '../services/randomService';
import { fetchAlternativeQuote } from '../services/alternativeService';

export const useQuote = () => {
    const [message, setMessage] = useState({} as Quote);

    useEffect(() => {
        const fetchQuote = async () => {
            try {
                const quote = await fetchRandomQuote();
                setMessage(quote);
            } catch (error) {
                try {
                    const alternativeQuote = await fetchAlternativeQuote();
                    setMessage(alternativeQuote);
                } catch (error) {
                    setMessage({
                        quote: `I'm sorry, Dave. I'm afraid I can't do that.`,
                        author: 'Hal 9000',
                    });
                }
            }
        };
        fetchQuote();
    }, []);

    return message;
};
