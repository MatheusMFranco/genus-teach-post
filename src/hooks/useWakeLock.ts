import { useEffect, useState } from 'react';

export const useWakeLock = () => {
    const [wakeLock, setWakeLock] = useState<WakeLockSentinel | null>(null);

    useEffect(() => {
        const enableWakeLock = async () => {
            try {
                const lock = await navigator.wakeLock.request('screen');
                setWakeLock(lock);
                lock.addEventListener('release', () => {});
            } catch (err) {
                console.error('Wake Lock request failed:', err instanceof Error ? err.message : err);
            }
        };

        enableWakeLock();

        return () => {
            if (wakeLock) {
                wakeLock.release().then(() => {});
            }
        };
    }, [wakeLock]);

    return wakeLock;
};
