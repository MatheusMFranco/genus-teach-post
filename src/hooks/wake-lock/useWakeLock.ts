import { useState, useEffect } from 'react';

export const useWakeLock = () => {
  const [wakeLock, setWakeLock] = useState<WakeLockSentinel | null>(null);

  useEffect(() => {
    const enableWakeLock = async () => {
      try {
        const lock = await navigator.wakeLock.request('screen');
        setWakeLock(lock);
      } catch (_) {
        console.error('Wake Lock request failed');
      }
    };
    enableWakeLock();
  });
  return wakeLock;
};
