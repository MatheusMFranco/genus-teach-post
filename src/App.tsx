import { useEffect, useState } from 'react';

import styles from './App.module.less';
import logo from './medias/ghost.jfif';

import { Quote } from './models/Quote';
import { useFullScreen } from './utils/useFullScreen';
import { fetchRandomQuote } from './services/quoteService';

function App() {
  const [message, setMessage] = useState({} as Quote);
  const { toggleFullScreen } = useFullScreen();

  useEffect(() => {
    fetchRandomQuote().then(setMessage);
  }, []);

  return (
    <article className={styles.quote} onClick={toggleFullScreen}>
      <div className={styles['quote-media']}>
        <img src={logo} className={styles['quote-picture']} alt="Zeitgeist" />
      </div>
      <blockquote className={styles['quote-box']}>
        <h1 className={styles['quote-message']}>
          {message.quote || '...'}
        </h1>
        <cite className={styles['quote-author']}>
          {message.author || 'Unknown'}
        </cite>
      </blockquote>
    </article>
  );
}

export default App;
