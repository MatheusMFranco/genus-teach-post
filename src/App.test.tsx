import { render, screen } from '@testing-library/react';

import App from './App';
import { useWakeLock } from './hooks/wake-lock/useWakeLock';
import { useFullScreen } from './hooks/full-screen/useFullScreen';
import { useQuote } from './hooks/quote/useQuote';
import { useAutoRefresh } from './hooks/auto-refresh/useAutoRefresh';

jest.mock('./hooks/wake-lock/useWakeLock', () => ({
  useWakeLock: jest.fn(),
}));
jest.mock('./hooks/full-screen/useFullScreen');
jest.mock('./hooks/auto-refresh/useAutoRefresh');
jest.mock('./hooks/quote/useQuote');

describe('<App />', () => {

  Object.defineProperty(navigator, 'wakeLock', {
    value: {
      request: jest.fn().mockResolvedValue({
        release: jest.fn(),
      }),
    },
  });  

  beforeEach(() => {
    (useFullScreen as jest.Mock).mockReturnValue({ toggleFullScreen: jest.fn() });
    (useWakeLock as jest.Mock).mockImplementation(() => {}); // Mock sem funcionalidade
    (useAutoRefresh as jest.Mock).mockImplementation(() => {}); // Mock sem funcionalidade
  });

  it('should render', () => {
    const quoteData = { quote: 'Test quote', author: 'Test author' };
    (useQuote as jest.Mock).mockReturnValue(quoteData);
    render(<App />);
    expect(screen.getByRole('heading')).toBeInTheDocument();
  });

  it('renders App component correctly', () => {
    const { asFragment } = render(<App />);
    expect(asFragment()).toMatchSnapshot();
  });
});
