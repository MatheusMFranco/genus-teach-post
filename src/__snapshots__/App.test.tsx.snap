// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<App /> renders App component correctly 1`] = `
<DocumentFragment>
  <article
    class="quote"
  >
    <div
      class="quote-media"
    >
      <img
        alt="Zeitgeist"
        class="quote-picture"
        src="ghost.jfif"
      />
    </div>
    <blockquote
      class="quote-box"
    >
      <h1
        class="quote-message"
      >
        ...
      </h1>
      <cite
        class="quote-author"
      >
        Unknown
      </cite>
    </blockquote>
  </article>
</DocumentFragment>
`;
